### PR TITLE
feat(openrouter): add model scout benchmark blueprint

### PR DESCRIPTION
**Pull request by chaliy**
## What

Adds the `openrouter_model_scout` AgentBlueprint contributed by a new `ModelScoutCapability`. The blueprint lets an orchestrating agent benchmark OpenRouter model/provider routes and produce an explicit model-router update proposal.

New files: `crates/core/src/capabilities/model_scout.rs`
Touched: `capabilities/mod.rs` (module + registration), `specs/agent-blueprints.md`

## Why

Without a scout workflow, model-router tuning is manual and stale. Operators have no in-product way to answer "which low-cost model can reliably use tools?" or "which provider is fastest for summarization?" This blueprint provides that capability as an agent-driven, auditable workflow with guardrails on spend and candidate count.

Fixes EVE-568.

## How

**`ModelScoutCapability`** — registered in `CapabilityRegistry::with_builtins()`, contributes one `AgentBlueprint`:

| Tool | Purpose |
|---|---|
| `list_openrouter_catalog` | Fetches OpenRouter `/api/v1/models` with filters (tools, JSON, price ceiling) |
| `probe_model` | Runs probe tasks against a candidate via OpenRouter's chat completions API; captures latency, success, token usage, check outcomes |
| `rank_models` | Aggregates results into per-model rankings; composite score: success_rate 50 %, latency 30 %, cost efficiency 20 % |
| `propose_router_update` | Generates a model-router route proposal (primary + fallbacks) from rankings |

Blueprint model: `claude-haiku-4-5-20251001` (cheap orchestrator; probe calls hit candidates directly). Config schema exposes `max_candidates` (≤ 50, default 10), `max_spend_usd` (default $0.10), `probe_timeout_ms`, custom `probe_tasks`, and `target_route_key`.

Ranking algorithm (`compute_score`): `success_rate × 0.50 + (1 − latency/15s) × 0.30 + (1 − cost_per_probe/max_cost) × 0.20`. Three built-in probe tasks (basic response, tool hint, JSON output) used when no custom tasks are provided.

## Risk

- **Low** — additive change; no schema migrations, no new API endpoints, no changes to existing runtime paths.
- Probe calls require an OpenRouter provider credential in `provider_credential_store`. Missing credentials return a clear tool error; no session-level failure.

## Security

Reviewed: TM-LLM, TM-DOS, TM-API

- **TM-LLM**: Probe prompts are fixed strings (from built-in tasks or operator config); sent only to OpenRouter endpoints. API key sourced from `provider_credential_store` and never logged or returned.
- **TM-DOS**: `max_candidates` (default 10, max 50) and `max_spend_usd` (default $0.10, max $10) bound probe count and spend. Per-probe timeout bounded to 60 s.
- **TM-API**: No new HTTP endpoints added.

No new THREAT comments required.

## Follow-ups

- **Expose `metadata` / `provider` routing in probe calls**: probe calls do not yet set OpenRouter provider constraints. A follow-up can thread `OpenRouterRoutingConfig` into probe requests to benchmark specific provider routes, not just models. (Out of scope for this PR which establishes the data model and ranking logic.)
- **Wire blueprint to OpenAPI spec**: blueprint tool definitions are not exposed in the OpenAPI spec yet (blueprints are runtime-only). Deferred pending a broader blueprint catalog endpoint.

## Checklist

- [x] Tests added or updated (11 unit tests covering ranking, scoring, edge cases, and tool execute paths)
- [x] Backward compatibility considered (additive; no existing behavior changed)
- [x] Security review performed against relevant threat model categories
- [x] Final CI ran checks affected by this PR
- [x] All review comments addressed (code change or written reasoning)

### CODE REVIEW

File: crates/core/src/capabilities/model_scout.rs [hunk 1]
```diff
@@ -0,0 +1,1160 @@
...
+        };
+
+        let models = match body.get("data").and_then(|d| d.as_array()) {
+            Some(arr) => arr,
+            None => {
+                return ToolExecutionResult::tool_error(
+                    "OpenRouter catalog response missing 'data' array".to_string(),
+                );
+            }
+        };
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/model_scout.rs:433]: Fixed in 09084e6. Models are now sorted by name before the filter+truncation chain, so `take(max_results)` always pulls from a stable ordering regardless of the API's response order.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/model_scout.rs [hunk 2]
```diff
@@ -0,0 +1,1160 @@
...
+                        .get("pricing")
+                        .and_then(|pr| pr.get("prompt"))
+                        .and_then(|v| v.as_str())
+                        .and_then(|s| s.parse().ok())
+                        .unwrap_or(0.0);
+                    // OpenRouter reports price per token; convert to per-million
+                    if prompt_price * 1_000_000.0 > max_p {
+                        return false;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/model_scout.rs:477]: Fixed in 09084e6. When `max_prompt_price_per_million` is active, models whose pricing field is missing or unparseable are now excluded (`None` branch returns `false`). Only models with a parseable price at or below the ceiling pass through.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/model_scout.rs [hunk 3]
```diff
@@ -0,0 +1,1160 @@
...
+
+        let api_key = match resolve_openrouter_key(context).await {
+            Ok(k) => k,
+            Err(e) => return ToolExecutionResult::tool_error(e),
+        };
+
+        let client = reqwest::Client::builder()
+            .timeout(std::time::Duration::from_millis(timeout_ms))
+            .build()
+            .unwrap_or_else(|_| reqwest::Client::new());
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/model_scout.rs]: Fixed in 09084e6. Client build failures now return a `tool_error` with the underlying message instead of silently falling back to an unconfigured client.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/model_scout.rs [hunk 4]
```diff
@@ -0,0 +1,1160 @@
...
+
+    let input_tokens = body
+        .get("usage")
+        .and_then(|u| u.get("prompt_tokens"))
+        .and_then(|v| v.as_u64());
+    let output_tokens = body
+        .get("usage")
+        .and_then(|u| u.get("completion_tokens"))
+        .and_then(|v| v.as_u64());
+
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/model_scout.rs:729]: Fixed in 09084e6. `run_probe` now extracts `body["usage"]["cost"]` and passes it as `cost_usd` into the returned `ProbeResult`, enabling the cost component of `compute_score`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/model_scout.rs [hunk 5]
```diff
@@ -0,0 +1,1160 @@
...
+    // Evaluate checks
+    let mut passed = vec![];
+    let mut failed = vec![];
+    for check in &task.checks {
+        let ok = match check.as_str() {
+            "not_empty" => !content.trim().is_empty(),
+            "max_latency_5s" => latency_ms <= 5_000,
+            "max_latency_10s" => latency_ms <= 10_000,
+            _ => true, // unknown checks pass by default
+        };
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/model_scout.rs:741]: Fixed in 09084e6. Unknown check IDs now map to `false` in the match arm with a comment explaining the intent. Typos in task configs will immediately show up as failed checks rather than inflating success rates.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/model_scout.rs [hunk 6]
```diff
@@ -0,0 +1,1160 @@
...
+        let top_n = arguments.get("top_n").and_then(|v| v.as_u64()).unwrap_or(3) as usize;
+
+        let top_models: Vec<String> = rankings
+            .iter()
+            .take(top_n)
+            .map(|r| r.model_id.clone())
+            .collect();
+
+        let best = &rankings[0];
+        let rationale = format!(
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/model_scout.rs]: Fixed in 09084e6. `propose_router_update` now sorts the incoming rankings by score descending before selecting `rankings[0]` and `.take(top_n)`, so callers don't need to pre-sort.

---
_Generated by [Claude Code](https://claude.ai/code)_
